### PR TITLE
Bump mypy[c] from 0.971 to 0.991

### PR DESCRIPTION
**Pull request by ichard26**
### Description

Fixes GH-3376.

Passed just fine: https://github.com/ichard26/black-mypyc-wheels/actions/runs/3452585255.

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary? -> n/a
- [x] Add new / update outdated documentation? -> n/a

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -35,6 +35,9 @@
 
 <!-- Changes to how Black is packaged, such as dependency requirements -->
 
+- Upgrade mypyc from `0.971` to `0.990` so mypycified _Black_ can be built on armv7
+  (#3380)
```
Comment (ichard26) [inline CHANGES.md:39]: Not sure if this is really necessary, armv7 is probably very minor.

**Review by JelleZijlstra — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (JelleZijlstra) [reply CHANGES.md:39]: Somebody asked for it so it will help some users :)

**Comment by ichard26** [comment]
@isac322, I'd appreciate it if you could check out this PR and verify Black builds on your armv7 machine.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -38,7 +38,7 @@ repos:
           - flake8-simplify
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.971
+    rev: v0.990
```
Comment (JelleZijlstra) [inline .pre-commit-config.yaml:41]:
```suggestion
    rev: v0.991
```

File: CHANGES.md [hunk 3]
```diff
@@ -35,6 +35,9 @@
 
 <!-- Changes to how Black is packaged, such as dependency requirements -->
 
+- Upgrade mypyc from `0.971` to `0.990` so mypycified _Black_ can be built on armv7
```
Comment (JelleZijlstra) [inline CHANGES.md:38]:
```suggestion
- Upgrade mypyc from `0.971` to `0.991` so mypycified _Black_ can be built on armv7
```

File: pyproject.toml [hunk 4]
```diff
@@ -119,7 +119,7 @@ sources = ["src"]
 enable-by-default = false
 dependencies = [
   "hatch-mypyc>=0.13.0",
-  "mypy==0.971",
+  "mypy==0.990",
```
Comment (JelleZijlstra) [inline pyproject.toml:122]:
```suggestion
  "mypy==0.991",
```

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**